docs/migration_guide.rst

```diff
@@ -1,10 +1,55 @@
+X.X.X migration guide
+=======================
+
+TODO(Update this with correct version or move to separate file)
+
+This version deprecates the AccountClient, which is a major change to the StarkNet.py
+It is being replaced with new :class:`BaseAccount <starknet_py.net.account.base_account.BaseAccount>` ABC and its
+default implementation :class:`Account <starknet_py.net.account.account.Account>`.
+
+Unlike ``AccountClient``, an ``Account`` is not a Client anymore. That means that methods like
+``get_storage_at``, ``call_contract`` etc. are not available in the Account interface.
+
+However ``Account`` now exposes a `.`client`` property, which means using an ``Account`` is
+just as simple as ``AccountClient`` was. For example:
+
+.. code-block:: python
+
+    account_client.get_storage_at(contract_address=0x1234)
+    # becomes
+    account.client.get_storage_at(contract_address=0x1234)
+
```

Comment:
I would add a few more examples of things that people did before (calling contracts, sending deploy_account, executing).